kin-ecosystem-sdk/src/main/java/com/kin/ecosystem/data/order/CreateExternalOrderCall.java

```diff
@@ -73,12 +82,15 @@ public void onChanged(final Payment payment) {
                     if (payment.isSucceed()) {
                         getOrder(payment.getOrderID());
                     } else {
-                        runOnMainThread(new Runnable() {
-                            @Override
-                            public void run() {
-                                externalOrderCallbacks.onTransactionFailed(openOrder, payment.getResultMessage());
-                            }
-                        });
+                        if (externalOrderCallbacks instanceof ExternalSpendOrderCallbacks) {
```

Comment:
you can use getOfferType() == OfferType.SPEND - it better to have one way of checking for readability